### PR TITLE
Expire experiments after 12 hours

### PR DESCRIPTION
**Pull request by balderdash**

### CODE REVIEW

File: apps/src/util/experiments.js [hunk 1]
```diff
@@ -24,23 +24,42 @@ experiments.getQueryString_ = function () {
...
-    allEnabled.push(key);
-    trackEvent(GA_EVENT, 'enable', key);
-  } else if (allEnabled.indexOf(key) >= 0 && !shouldEnable) {
-    allEnabled.splice(allEnabled.indexOf(key), 1);
+  const allEnabled = this.getStoredExperiments_();
+  const experimentIndex =
+    allEnabled.findIndex(experiment => experiment.key === key);
+  if (shouldEnable) {
+    const expirationDate = new Date();
+    expirationDate.setHours(expirationDate.getHours() + 12);
```
Comment (joshlory) [inline apps/src/util/experiments.js]: Nit: magic number.

File: apps/src/util/experiments.js [hunk 2]
```diff
@@ -24,23 +24,42 @@ experiments.getQueryString_ = function () {
...
 experiments.setEnabled = function (key, shouldEnable) {
-  const allEnabled = this.getEnabledExperiments();
-  if (allEnabled.indexOf(key) < 0 && shouldEnable) {
-    allEnabled.push(key);
-    trackEvent(GA_EVENT, 'enable', key);
-  } else if (allEnabled.indexOf(key) >= 0 && !shouldEnable) {
-    allEnabled.splice(allEnabled.indexOf(key), 1);
+  const allEnabled = this.getStoredExperiments_();
+  const experimentIndex =
+    allEnabled.findIndex(experiment => experiment.key === key);
```
Comment (joshlory) [inline apps/src/util/experiments.js:51]: Does this get polyfilled by our build process?  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex#Browser_compatibility

File: apps/test/unit/experimentTest.js [hunk 3]
```diff
@@ -83,4 +102,45 @@ describe('experiments', function () {
...
+    };
+
+    localStorage.setItem('experimentsList',
+        JSON.stringify(['awesome-feature']));
+    assert.isFalse(experiments.isEnabled('awesome-feature'));
+
+    localStorage.getItem = originalGetItem;
+  });
+
+  it('expires old-style experiments in localstorage', function () {
```
Comment (joshlory) [inline apps/test/unit/experimentTest.js:140]: 👍

File: apps/src/util/experiments.js (hunk 2 above)
Comment (balderdash) [reply apps/src/util/experiments.js:51]: Yep, and I just manually tested on IE to be sure.

**Pull request merged**